azurerm/internal/services/servicebus/resource_arm_servicebus_subscription.go

```diff
@@ -145,6 +158,7 @@ func resourceArmServiceBusSubscriptionCreateUpdate(d *schema.ResourceData, meta
 			EnableBatchedOperations:          &enableBatchedOps,
 			MaxDeliveryCount:                 &maxDeliveryCount,
 			RequiresSession:                  &requiresSession,
+			Status:                           helper.ExpandEntityStatus(d.Get("status")),
```

Comment:
I appreciate all the work you put into making the helper function to get the status but we should aim for simplicity. We know that `status` can only be `servicebus.EntityStatus` because of what we allow in the  `ValidateFunc` so we can be confident that we can get away with a single line versus the helper function + tests that you wrote

```suggestion
			Status:                           servicebus.EntityStatus(d.Get("status").(string)),
```

Reply:
Changed this to a one-liner in 87b1115c70dd1df99865cd7288d0627b7c751842